src/VCR.ts

```diff
@@ -12,30 +12,33 @@ interface VCROptions {
 export async function startVCR(options?: Partial<VCROptions>): Promise<nock.Scope[]> {
   const { processDefns, CI } = { processDefns: defns => defns, CI: process.env.CI, ...options } as VCROptions
   const defns = await currentCassettes()
-  if (!defns.length) {
-    if (CI) {
-      throw new Error(`No cassettes found. They must be in place before running tests on CI ${cassettePath()}`)
-    }
-    // No cassettes found - recording responses
-    nock.recorder.rec({ output_objects: true, dont_print: true })
-  } else {
+
+  if (CI && typeof defns === 'undefined') {
+    throw new Error(`No cassettes found. They must be in place before running tests on CI ${cassettePath()}`)
+  }
+
+  if (defns && !!defns.length) {
     // Cassettes found - using saved responses
     // set up the mocks
     const scopes = nock.define(processDefns(defns.map(relaxDefinitionsForJsonRPC)))
     scopes.forEach(matchJsonRPCResponseToRequest)
     if (!nock.isActive()) nock.activate()
     return scopes
   }
+
+  const shouldRecord = !CI && (typeof defns === 'undefined' || defns.length === 0)
```

Comment:
It's really a nitpick at this point so I won't hold this up anymore, but if we hit this line, we know that defns is either undefined or empty (because it does not contain items).